/home/ritwik/.local/lib/python3.8/site-packages/gym/spaces/box.py:127: UserWarning: [33mWARN: Box bound precision lowered by casting to float32
  logger.warn(f"Box bound precision lowered by casting to {self.dtype}")
[33m[WARN] [1724954015.850081, 0.009000]: Start Init ControllersConnection
[33m[WARN] [1724954015.851805, 0.009000]: END Init ControllersConnection
Entered stewart pose Env
Entered Stewart Env
launch the init function of the Parent Class robot_gazebo_env_goal.RobotGazeboEnv.....
launch the init function of the Parent Class robot_gazebo_env_goal.RobotGazeboEnv.....END
check_all_sensors_ready checked..
[34m[1mwandb[39m[22m: Network error resolved after 0:02:48.307911, resuming normal operation.
[31m[ERROR] [1724989136.328286, 49.095000]: bad callback: <bound method StewartEnv.links_callback of <reaching_pose_env.PoseSetEnv object at 0x7f96123c2f40>>
[31mTraceback (most recent call last):
[31m  File "/opt/ros/noetic/lib/python3/dist-packages/rospy/topics.py", line 750, in _invoke_callback
[31m    cb(msg)
[31m  File "/home/ritwik/catkin_ws/src/stewart_platform_learning/src/stewart_platform/scripts/stewart_env.py", line 150, in links_callback
[31m    self._pose_pub.publish(self.pose_msg)
[31m  File "/opt/ros/noetic/lib/python3/dist-packages/rospy/topics.py", line 882, in publish
[31m    self.impl.publish(data)
[31m  File "/opt/ros/noetic/lib/python3/dist-packages/rospy/topics.py", line 1041, in publish
[31m    raise ROSException("publish() to a closed topic")
[31mrospy.exceptions.ROSException: publish() to a closed topic
Traceback (most recent call last):
  File "/home/ritwik/catkin_ws/src/stewart_platform_learning/src/stewart_platform/scripts/DDPG_Continuous.py", line 237, in <module>
    main()
  File "/home/ritwik/catkin_ws/src/stewart_platform_learning/src/stewart_platform/scripts/DDPG_Continuous.py", line 231, in main
    env = gym.make(env_name)
  File "/home/ritwik/.local/lib/python3.8/site-packages/gym/envs/registration.py", line 640, in make
    env = env_creator(**_kwargs)
  File "/home/ritwik/catkin_ws/src/stewart_platform_learning/src/stewart_platform/scripts/reaching_pose_env.py", line 36, in __init__
    super(PoseSetEnv, self).__init__()
  File "/home/ritwik/catkin_ws/src/stewart_platform_learning/src/stewart_platform/scripts/stewart_env.py", line 57, in __init__
    self.gazebo.pauseSim()
  File "/home/ritwik/catkin_ws/src/stewart_platform_learning/src/openai_ros/openai_ros/src/openai_ros/gazebo_connection.py", line 35, in pauseSim
    rospy.wait_for_service('/gazebo/pause_physics')
  File "/opt/ros/noetic/lib/python3/dist-packages/rospy/impl/tcpros_service.py", line 166, in wait_for_service
    raise ROSInterruptException("rospy shutdown")
rospy.exceptions.ROSInterruptException: rospy shutdown
Traceback (most recent call last):
  File "/home/ritwik/catkin_ws/src/stewart_platform_learning/src/stewart_platform/scripts/DDPG_Continuous.py", line 237, in <module>
    main()
  File "/home/ritwik/catkin_ws/src/stewart_platform_learning/src/stewart_platform/scripts/DDPG_Continuous.py", line 231, in main
    env = gym.make(env_name)
  File "/home/ritwik/.local/lib/python3.8/site-packages/gym/envs/registration.py", line 640, in make
    env = env_creator(**_kwargs)
  File "/home/ritwik/catkin_ws/src/stewart_platform_learning/src/stewart_platform/scripts/reaching_pose_env.py", line 36, in __init__
    super(PoseSetEnv, self).__init__()
  File "/home/ritwik/catkin_ws/src/stewart_platform_learning/src/stewart_platform/scripts/stewart_env.py", line 57, in __init__
    self.gazebo.pauseSim()
  File "/home/ritwik/catkin_ws/src/stewart_platform_learning/src/openai_ros/openai_ros/src/openai_ros/gazebo_connection.py", line 35, in pauseSim
    rospy.wait_for_service('/gazebo/pause_physics')
  File "/opt/ros/noetic/lib/python3/dist-packages/rospy/impl/tcpros_service.py", line 166, in wait_for_service
    raise ROSInterruptException("rospy shutdown")
rospy.exceptions.ROSInterruptException: rospy shutdown